c++/src/capnp/serialize.c++

```diff
@@ -271,6 +271,11 @@ void readMessageCopy(kj::InputStream& input, MessageBuilder& target,
 
 void writeMessage(kj::OutputStream& output, kj::ArrayPtr<const kj::ArrayPtr<const word>> segments) {
   KJ_REQUIRE(segments.size() > 0, "Tried to serialize uninitialized message.");
+#ifdef _WIN32
+#ifndef NDEBUG
```

Comment:
Why does debug mode matter here?

Reply:
Just a tiny optimisation. In release mode, the `requireIsInBinaryMode()` call is a no-op anyway, but this removes even the call itself.

Reply:
Oh my bad. I misread this as saying this optimization is live only in release builds which confused me.